### PR TITLE
Tag flaky tests

### PR DESCRIPTION
**Pull request by emord**
Tagging some flaky tests found in https://github.com/dimagi/commcare-hq/pull/24735

@nickpell one of these is Odata, so I think it was added fairly recently. Maybe you have still have some context that could help fix it more permanently?

### CODE REVIEW

**Review by esoergel — APPROVED** [review]

**Review by kaapstorm — APPROVED** [review]

**Comment by millerdev** [comment]
I'm conflicted about using `@flaky` to "fix" intermittently failing tests. I maintain a document of tests that fail intermittently and do intend to work on them at some point, although admittedly I have not acted on that in quite a long time—hence the conflicted feelings.

I'd especially caution against using this on tests that have been written fairly recently since hopefully the person who wrote them should have reasonable context to rewrite them so they are no longer flaky.

Ultimately the accumulation of `@flaky` tests will add time to our test runs and potentially hide bugs from us. This is not an aspect of good testing culture.

**Comment by esoergel** [comment]
@millerdev I commented similarly on a recent PR.  Couple thoughts:

`@flaky` is _itself_ a way of keeping track of which tests fail frequently.  It calls attention to the fact that the test frequently fails, rather than obscuring it.

It doesn't make sense to require the authors of random PRs to fix unrelated test failures as a precondition for their PR to go through, at least not on the very same PR.

Given that, the alternative to `@flaky` is our current approach of hitting rebuild one or more times, re-running an entire test job multiple times because of one flaky test.  This is likely much worse for our build through-puts than a small number of tests running twice.

I also feel a "we should suffer the consequences of our sloppiness" impulse, but from a practical standpoint I don't think it has any advantages over this approach.  It doesn't seem we as a team have felt compelled by it to make urgent progress on test consistency.

I do agree that we should show restraint in its use, especially for new-ish PRs, and consider `@flaky` only a baby step above `@skip` - a hack to avoid freezing unrelated work, not an actual fix.

**Comment by millerdev** [comment]
@esoergel I think we are mostly on the same page. I agree that marking tests with `@flaky` clearly identifies those tests, which is a good thing. Your point about the cost/benefit trade-off of restarting the build manually vs automatic retry one test is also very good, and I agree completely.

The danger I see with adopting `@flaky` is that we will not only be lazy about flaky tests (as history shows we are wont to do), but allow such a pile of them to accumulate that it will actually make things worse in the long term. Maybe at that point the dev team will have a "fix flaky tests" day (or week) and stabilize enough of them to allow us to resume business as usual for a while? (rhetorical question) I suppose that's at least pragmatic.

One clarification:

> It doesn't make sense to require the authors of random PRs to fix unrelated test failures as a precondition for their PR to go through, at least not on the very same PR.

I'm not sure what I said that implied that a PR author should be responsible for fixing intermittent test failures that are _unrelated_ to her new PR. Or maybe I'm interpreting what you're saying incorrectly? However, if a new PR **causes** other, _seemingly unrelated_, tests to fail then I think it is reasonable to say the author of that PR is responsible for fixing the issue before merging. That may be hard to determine though.

**Comment by millerdev** [comment]
Also, to be clear, I am not blocking this PR from being merged.

**Comment by kaapstorm** [comment]
> Maybe at that point the dev team will have a "fix flaky tests" day

:smirk: Summit is in three months. ... Or flaky tests could also be "Interrupt"able.

**Pull request merged**
